lib/yggdrasil/plug/SkipIfAuthenticated.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Yggdrasil.Plug.SkipIfAuthenticated do
+  import Plug.Conn
+  import Yggdrasil.Router.Helpers
+  import Phoenix.Controller
+
+  def init(default), do: default
+
+  def call(conn, default) do
```

Comment:
Yes we should be able to test this as well, there are helpers as said below in another comment that would let you basically take a conn do something like get conn "/" so it sets it up for that path, then you can pipe it into the plug, so you can set up the sessions stuff I think as well.